Tests/TestingTests/Runner.RuntimeStateTests.swift

```diff
@@ -25,4 +25,21 @@ struct Runner_RuntimeStateTests {
 
     await Test(name: "foo") {}.run(configuration: configuration)
   }
+
+  // This confirms that an event posted within the closure of `withTaskGroup`
+  // (and similar APIs) in regular tests (those which are not testing the
+  // testing library itself) are handled appropriately and do not crash.
+  @Test func eventPostingInTaskGroup() async {
```

Comment:
Can we have an additional test (maybe written with XCTest) that sanity-checks that the state is `nil` by default?

Reply:
Sure, added that